web/src/components/SecondaryNav.js

```diff
@@ -1,77 +1,64 @@
 import React, { Fragment } from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
-import { Link } from 'react-router-dom';
-import { LinksContextConsumer } from '../contexts/LinksContextProvider';
-import NextPreviousButtons from './NextPreviousButtons';
-import { jumpTo } from '../actions/app';
+import { Link, useParams as actualUseParams } from 'react-router-dom';
+import ContinuePreviousButtons from './ContinuePreviousButtons';
 import {
   selectActivitiesSidebar,
   selectActivityCount
 } from '../reducers/activities.selectors';
-import { addActivity } from '../actions/editActivity';
-import { selectActiveSection } from '../reducers/navigation';
+import { addActivity as actualAddActivity } from '../actions/editActivity';
 
 const SecondaryNav = ({
-  activeSection,
-  activities,
   activityCount,
-  jumpTo: jumpAction,
-  add
+  addActivity,
+  location,
+  useParams
 }) => {
-  const onAdd = () => {
-    add();
-    jumpAction(`activities-list`);
-    window.scrollTo(0, 0);
-  };
-
-  const activityIndex = activities.findIndex(a =>
-    activeSection.includes(a.anchor)
-  );
+  const { activityIndex } = useParams();
+  const showAddActivityLink = activityIndex + 1 === activityCount &&
+    location.pathname.endsWith('ffp');
 
   return (
-    <LinksContextConsumer>
-      {context => (
-        <Fragment>
-          {activityIndex + 1 === activityCount &&
-            activeSection.includes('ffp') && (
-              <div className="pre-button-section-break">
-                <Link
-                  to="/apd/activities"
-                  onClick={onAdd}
-                  className="ds-c-button"
-                >
-                  Add another activity
-                </Link>
-              </div>
-            )}
-          <div className="pre-button-section-break">
-            <NextPreviousButtons context={context} />
-          </div>
-        </Fragment>
+    <Fragment>
+      {showAddActivityLink && (
+        <div className="pre-button-section-break">
+          <Link
+            to="/apd/activities"
+            onClick={addActivity}
+            className="ds-c-button"
+          >
+            Add another activity
+          </Link>
+        </div>
       )}
-    </LinksContextConsumer>
+      <div className="pre-button-section-break">
+        <ContinuePreviousButtons />
+      </div>
+    </Fragment>
   );
 };
 
+SecondaryNav.defaultProps = {
+  useParams: actualUseParams
+}
```

Comment:
This turned out to be an easy way to pass the `useParams` function from `react-router-dom` by default, and override it when testing.